src/components/BounceRuleDetailed/Modals/index.jsx

```diff
@@ -0,0 +1,188 @@
+          className="sg-button"
+          onClick={handleModalClose}
+          id={modalType}
+          type="secondary"
+        >
+          {"Close"}
+        </Button>
+      </Column>
+      <Column width={1} offset={11}>
+        <Button
+          className="sg-button"
+          id={modalType}
+          onClick={handleModalConfirm}
+          type="primary"
+        >
+          {"Confirm"}
+        </Button>
+      </Column>
+    </Row>
+  </div>
+);
+
+const ChangeModalBody = ({ currentRule, handleModalClose }) => (
+  <div className="changelog-modal">
+    <Row>
+      <Column width={6} offset={1}>
+        <h1>Current</h1>
+        <ChangeTable currentRule={currentRule} />
+      </Column>
+      <Column width={6} offset={7}>
+        <h1>Previous</h1>
+        <ChangeTable currentRule={currentRule} />
+      </Column>
+    </Row>
+    <Row>
+      <Column className="changelog-modal-button sg-right" width={1} offset={12}>
+        <Button id="changeModal" onClick={handleModalClose}>
+          Close
+        </Button>
+      </Column>
+    </Row>
+  </div>
+);
+
+const ChangeTable = ({ currentRule }) => {
+  const {
+    description,
+    id,
+    response_code: responseCode,
+    enhanced_code: enhancedCode,
+    bounce_action: bounceAction,
+    regex,
+    priority,
+  } = currentRule;
+  return (
+    <Table className="change-table">
+      <TableBody>
+        <TableRow>
+          <TableCell>
+            <strong>ID</strong>
+          </TableCell>
+          <TableCell>{id}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Bounce Action</strong>
+          </TableCell>
+          <TableCell>{bounceAction}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Response Code</strong>
+          </TableCell>
+          <TableCell>{responseCode}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Enhanced Code</strong>
+          </TableCell>
+          <TableCell>{enhancedCode}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>RegEx</strong>
+          </TableCell>
+          <TableCell>{regex}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Priority</strong>
+          </TableCell>
+          <TableCell>{priority}</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>
+            <strong>Description</strong>
+          </TableCell>
+          <TableCell>{description}</TableCell>
+        </TableRow>
+      </TableBody>
+    </Table>
+  );
+};
+
+const ModalDisplay = ({
+  isChangeModalOpen,
+  isConfirmOpen,
+  isCancelConfirmOpen,
+  currentRule,
+  handleModalClose,
+  handleModalConfirm,
+}) => (
+  <CenterModal
+    large={isChangeModalOpen}
+    open={isChangeModalOpen || isConfirmOpen || isCancelConfirmOpen}
+    renderBody={
+      (isChangeModalOpen && (
```

Comment:
Hmm why not create a function like renderBody that encapsulates all this and pass in renderBody()? and let's try to avoid doing all these ||s when you can do stuff like {condition && component} {condition && component} etc.